distribution/lib/Standard/Microsoft/0.0.0-dev/src/Internal/SQLServer_Connection.enso

```diff
@@ -0,0 +1,299 @@
+private
+
+from Standard.Base import all
+import Standard.Base.Metadata.Display
+from Standard.Base.Metadata.Choice import Option
+from Standard.Base.Metadata.Widget import Single_Choice
+
+import Standard.Table.Rows_To_Read.Rows_To_Read
+from Standard.Table import Table
+
+import Standard.Database.Column_Description.Column_Description
+import Standard.Database.Connection.Connection.Connection
+import Standard.Database.DB_Table.DB_Table
+import Standard.Database.Internal.Connection.Entity_Naming_Properties.Entity_Naming_Properties
+import Standard.Database.Internal.JDBC_Connection
+import Standard.Database.SQL_Query.SQL_Query
+import Standard.Database.SQL_Statement.SQL_Statement
+from Standard.Database.Connection.Connection import make_database_selector, make_schema_selector, make_structure_creator, make_table_name_selector, make_table_types_selector
+from Standard.Database.Errors import SQL_Error, Table_Already_Exists, Table_Not_Found
+from Standard.Database.Internal.Upload_Table import first_column_name_in_structure
+
+import project.Internal.SQLServer_Dialect
+
+type SQLServer_Connection
+    ## PRIVATE
+
+       Creates a SQLServer connection based on a URL and properties.
+
+       Arguments:
+       - url: The URL to connect to.
+       - properties: A vector of properties for the connection.
+       - make_new: A function that returns a new connection.
+    create : Text -> Vector -> (Text -> Text -> SQLServer_Connection) -> SQLServer_Connection
+    create url properties make_new =
+        jdbc_connection = JDBC_Connection.create url properties
+        entity_naming_properties = Entity_Naming_Properties.from_jdbc_connection jdbc_connection is_case_sensitive=False
+        SQLServer_Connection.Value (Connection.new jdbc_connection SQLServer_Dialect.sqlserver entity_naming_properties) make_new
+
+    ## PRIVATE
+
+       A SQLServer database connection.
+
+       Arguments:
+       - connection: the underlying connection.
+       - make_new: a function that returns a new connection.
+    Value connection make_new
+
+    ## ICON data_input
+       Closes the connection releasing the underlying database resources
+       immediately instead of waiting for them to be automatically released.
+
+       The connection is not usable afterwards.
+    close : Nothing
+    close self = self.connection.close
+
+    ## ICON metadata
+       Returns the list of warehouses for the connection.
+    warehouses : Vector Text
+    warehouses self =
+        self.connection.read_single_column "show warehouses" "name"
+
+    ## ICON metadata
+       Returns the name of the current warehouse.
+    warehouse : Text
+    warehouse self =
+        vector = self.connection.read_single_column "SELECT current_warehouse() as name" "name"
+        vector.first
+
+    ## ICON data_input
+       Returns a new Connection with the specified database set as default.
```

Comment:
```suggestion
       Returns a new Connection with the specified warehouse set as default.
```

Should this comment be updated?

Reply:
No such warehouse in SQL Server - guessing this came from snowflake